docs/developer_guide.md

```diff
@@ -0,0 +1,59 @@
+# Setup Envoy AI Gateway Local Development Environment
+
+## Prerequisites
+- Go 1.23 or later
+- Envoy proxy installed
+- OpenAI API or AWS Bedrock key
```

Comment:
what exactly ` AWS Bedrock key` will be the obvious question for anyone coming here without context.